### PR TITLE
CASMCMS-7676 Add support for recipes with template params

### PR DESCRIPTION
**Pull request by ecozzi-hpe**
## Summary and Scope

Enable IMS python helper to add recipes to IMS that include a template-dictionary

_Is this change backwards incompatible, backwards compatible, or a backwards compatible bugfix?_

## Issues and Related PRs

* Resolves [CASMCMS-7676](https://jira-pro.its.hpecorp.net:8443/browse/CASMCMS-7676)

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * `<development system>`
  * Local development environment
  * Virtual Shasta

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable
- [ ] Do not merge until CSMv2.5 code is able to be merged

### CODE REVIEW

File: ims_python_helper/__init__.py [hunk 1]
```diff
@@ -357,7 +358,7 @@ def image_set_job_status(self, ims_job_id, job_status):
             ),
         }
 
-    def recipe_upload(self, name, filepath, distro):
+    def recipe_upload(self, name, filepath, distro, template_dictionary):
```
Comment (rkleinman-hpe) [inline ims_python_helper/__init__.py:361]: Will this be a major revision? This is a backward incompatible change. Also, missing explanation of this new arg in the docstring below (line 375ish)

File: ims_python_helper/__init__.py [hunk 2]
```diff
@@ -478,13 +479,19 @@ def _ims_recipe_create(self, name, linux_distribution):
...
-            "POST %s name=%s, linux_distribution=%s",
-            name, url, linux_distribution
+            "POST %s name=%s, linux_distribution=%s, template_dictionary=%s",
+            name, url, linux_distribution, template_dictionary
         )
+
         body = {
             'recipe_type': 'kiwi-ng',
             'linux_distribution': linux_distribution,
+            'template_dictionary': template_dictionary,
```
Comment (rkleinman-hpe) [inline ims_python_helper/__init__.py:494]: Hopefully no one puts any sensitive information in this field. Maybe worth a disclaimer in docs that this data is available to anyone with access to the IMS API.

**Review by rkleinman-hpe — CHANGES_REQUESTED** [review]

File: ims_python_helper/__init__.py (hunk 1 above)
Comment (ecozzi-hpe) [reply ims_python_helper/__init__.py:361]: Good catch. I'll make the extra parameter optional and conditionalize its use elsewhere.

File: ims_python_helper/__init__.py (hunk 2 above)
Comment (ecozzi-hpe) [reply ims_python_helper/__init__.py:494]: Yeah, this is not meant for sensitive information.

**Review by rkleinman-hpe — APPROVED** [review]

**Review by mharding-hpe — APPROVED** [review]

**Pull request closed without merging**